### ISSUE TITLE
connected service doesn't handle nillable date's

### ISSUE THREAD

**Issue by KLuuKer** [dotnet/wcf#2547]
i have a old legacy "Service Reference" style wcf client that is working with this wsdl correctly (i made sure it was updated for the current published wsdl)
and when trying to convert all the old legacy stuff to the new "Connected Services" it fails to properly handle because the wsdl has Nullable<DateTime>'s inside (thanks java....)

wsdl https://ws.tradetracker.com/soap/merchant?wsdl

**repo code**
```
// setup the client
var soapClient = new MerchantInterfaceClient(
new BasicHttpBinding(BasicHttpSecurityMode.Transport) { AllowCookies = true, MaxReceivedMessageSize = 1024 * 10 * 10 * 10 },
new EndpointAddress(new Uri("https://ws.tradetracker.com/soap/merchant")));

// replace "nl_NL" with "Locale.nl_NL" when using the old net461 wcf service reference
var tsk = soapClient.authenticateAsync("123456", "12345abcdefghijklmnopqrstuvwxyz123456789", true, "nl_NL", false);
tsk.Wait();

```
**expected result** (because these credentials are fake of course)

> FaultException: Failed to authenticate.

**actual result**
`System.InvalidOperationException: ''date' is an invalid value for the SoapElementAttribute.DataType property. The property may only be specified for primitive types.'`

**Comment by mlacouture** [comment]
Hi @KLuuKer 
Thank you for reporting this issue, unfortunately it is an known issue in the framework for which a fix has been rejected considering the risk and the scenario.  One of my colleagues who investigated long time ago gave the following reason:

> This bug only occurs when using SOAP encoding. SOAP encoding is not Basic Profile compliant (see http://www.ws-i.org/Profiles/BasicProfile-1.1-2004-08-24.html). The Basic Profile defines a set of guidelines and constraints for web services that improve interoperability across platforms. Unfortunately, because this bug only applies to legacy services that are not Basic Profile compliant it is unlikely that this bug will get fixed.

This issue has been discussed in the MSDN Forums and there is a[ proposed workaround](https://social.msdn.microsoft.com/Forums/en-US/3880b508-2a7b-4d0b-bb02-6b179c8153bb/net-bug-error-reflecting-complex-object-with-nullabledatetime?forum=asmxandxml).  Please notice that I have not tested this workaround. 

hope this helps,
Miguel Lacouture

**Comment by KLuuKer** [comment]
Hi @mlacouture i think you didn't understand me correctly (i know you guy's have allot to read)

**TLDR**
> - The soap client is currently **working** when created **using** the old **Service Reference**.
> - It's **NOT working** when **using** the **new** **Connected Services**
> 

Was it perhaps secretly fixed in some visualstudio\\.net framework combo?

I have also tried the following workaround's but they don't work:
[System.Xml.Serialization.SoapElementAttribute(DataType="date", IsNullable=true)]  
[System.Xml.Serialization.SoapElementAttribute(DataType="dateTime", IsNullable=true)]  
[System.Xml.Serialization.SoapElementAttribute(DataType="date")]  
[System.Xml.Serialization.SoapElementAttribute(DataType="dateTime")] [System.Xml.Serialization.SoapElementAttribute(IsNullable=true)]  

on these types of property's (yes i made sure all property's where modified between test runs)
public System.Nullable<System.DateTime> Date { ...... }

**Comment by mlacouture** [comment]
Hi @KLuuKer! I see, thank you for clarifying.  We'll take a look ...
